### PR TITLE
[SYCL] Enable xpti::node_create signal emit for parallel_for that bypasses graph

### PR DESCRIPTION
**Pull request by KseniyaTikhomirova**
xpti::node_create signal is emitted when we create new node in graph. Code related to it is present in Command::emitInstrumentationData and Command successors. Although we have a path when no memory dependencies is tracked for kernel (e.g. queue::parallel_for) and to speed up kernel enqueue and eliminate extra overhead - node is not added to graph (and related Command is not created too). This commit adds this node_create signal to be emitted in this case.

### CODE REVIEW

**Comment by KseniyaTikhomirova** [comment]
@intel/llvm-reviewers-runtime could you please review?

**Comment by KseniyaTikhomirova** [comment]
@cperkinsintel , @intel/llvm-reviewers-runtime kindly reminder, could you please review?

**Review by steffenlarsen — APPROVED** [review]
Apart from the small comments, these changes LGTM!

File: sycl/source/detail/scheduler/commands.cpp [hunk 1]
```diff
@@ -1851,51 +1851,116 @@ ExecCGCommand::ExecCGCommand(std::unique_ptr<detail::CG> CommandGroup,
   emitInstrumentationDataProxy();
 }
 
-void ExecCGCommand::emitInstrumentationData() {
+std::string instrumentationGetKernelName(
+    const std::shared_ptr<detail::kernel_impl> &SyclKernel,
+    const std::string &FunctionName, const std::string &SyclKernelName,
+    void *&Address, std::optional<bool> &FromSource) {
```
[the remark this answers is not in the document]
Comment (KseniyaTikhomirova) [reply sycl/source/detail/scheduler/commands.cpp:1858]: wrapped free functions with @ifdef and added std::ignore to the Command api function

[the remark this answers is not in the document]
Comment (steffenlarsen) [reply sycl/source/detail/scheduler/commands.cpp:1858]: Works for me, thank you!

**Pull request merged**